### DISCUSSION TITLE
Show category name just once

### DISCUSSION THREAD

**Discussion by Gw3i** [neuefische/questions#104] [Web] [answered]
Hello everyone!
I want to render my category page and just show categories with videos inside.
If I filter and map, the category name is displayed for each video, but I want to display it just once.
So if I have 2 video, the category name will displayed twice, when 3 video then 3 times and so on...

Here is a Screenshot:
<img width="1434" alt="Screenshot 2022-08-24 at 10 47 17" src="https://user-images.githubusercontent.com/101993818/186375853-9c5677e8-d695-4bf1-97fe-23a54ce93915.png">

Here is my current code:  https://github.com/Gw3i/capstone-project/pull/24
Here is the Preview: https://capstone-project-o5q2j7uq8-gw3i.vercel.app/explore

Thanks in advance!

**Comment by ahohnsen** [comment]
Hey Vlady,

You could achieve this using `Set`. Have a look at this article: https://dev.to/clairecodes/how-to-create-an-array-of-unique-values-in-javascript-using-sets-5dg6

**Reply by Gw3i** [reply]
Hey @ahohnsen,
thanks for the quick answer!

I've tried it, but not quite sure if it's right that way.
Unfortunately ```let categoryName = [...new Set(categoryNames)]```
 is returning me an empty array although ```const categoryNames = []``` has the category names inside it.
 Do you see the issue?
<img width="1426" alt="Screenshot 2022-08-24 at 11 39 16" src="https://user-images.githubusercontent.com/101993818/186386612-31ceade9-4bf4-432e-aa12-a298089d7906.png">
<img width="1420" alt="Screenshot 2022-08-24 at 11 41 55" src="https://user-images.githubusercontent.com/101993818/186386652-8fba9836-75e8-43a1-80dc-4fc1c6388d0e.png">

**Comment by ahohnsen** [comment, marked as answer]
Right now you are mapping over your full category list you have saved in your store instead of the categories which are really in use. I would solve the problem with the following steps:

- Map over all videos and return the video category to create an array with all used categories e.g. `['science', 'science', 'biology', 'art', 'biology']`
- Create an array of the different category names with `Set` - have a look at the article above. You should get an array with only unique values e.g. `['science', 'biology', 'art']`
- Map over that array and return for each category the category name and the videos with that category tag
